### PR TITLE
sssd python3

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
- Related ticket: https://progress.opensuse.org/issues/50486
- Verification runs:
  - Passed: http://slindomansilla-vm.qa.suse.de/tests/1396
  - Missing systemctl command: http://slindomansilla-vm.qa.suse.de/tests/1397#step/sssd/12
  - Missing service name: http://slindomansilla-vm.qa.suse.de/tests/1399#step/sssd/12

### CODE REVIEW

File: lib/consoletest.pm [hunk 1]
```diff
@@ -6,8 +21,14 @@ use warnings;
 use testapi;
 use known_bugs;
 
-# Base class for all console tests
+######################################
```
Comment (foursixnine) [inline lib/consoletest.pm]: Better use pod: 
```suggestion
=head 1 NAME

<consoletest> - Base class for all console tests

=cut
```

File: lib/Utils/Systemctl.pm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package Utils::Systemctl;
```
Comment (foursixnine) [inline lib/Utils/Systemctl.pm]: I would expect this namespace to manage systemd related stuff, not just systemctl :)

File: lib/Utils/Systemctl.pm [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+
+use utils 'systemctl';
```
Comment (foursixnine) [inline lib/Utils/Systemctl.pm]: I'd move systemctl to this package also.

File: lib/consoletest.pm [hunk 4]
```diff
@@ -6,8 +21,14 @@ use warnings;
...
+## Base class for all console tests ##
+######################################
+
 
+=head2 post_run_hook
```
Comment (foursixnine) [inline lib/consoletest.pm]: Please add a whitespace (Makes formatting nicer when calling perldoc foo.pm)

```suggestion
=head2 post_run_hook

```

File: lib/consoletest.pm [hunk 5]
```diff
@@ -18,6 +39,10 @@ sub post_run_hook {
     $self->clear_and_verify_console;
 }
 
+
+=head2 post_fail_hook
```
Comment (foursixnine) [inline lib/consoletest.pm]: same as above

File: lib/Utils/Systemctl.pm (hunk 3 above)
Comment (SergioAtSUSE) [reply lib/Utils/Systemctl.pm]: Yes, but when I was editing the file number 186 (and this without verification runs and without checking if the general `use utils;` was still necessary after adding use `Utils::Systemd;'...) and I saw that the scroll bar wasn't even in the middle, I thought of creating another ticket for that. We have to save the world another day. I have kept the changes on my git stash.

**Comment by SergioAtSUSE** [comment]
Follow up for saving the world: https://progress.opensuse.org/issues/51224

File: lib/consoletest.pm [hunk 6]
```diff
@@ -1,13 +1,40 @@
...
 use known_bugs;
 
-# Base class for all console tests
 
+=head1 consoletest
+
+<consoletest> - Base class for all console tests
+
+=cut
+
```
Comment (foursixnine) [inline lib/consoletest.pm]: Extra line

File: tests/console/cups.pm [hunk 7]
```diff
@@ -14,7 +14,8 @@ use base 'consoletest';
 use strict;
 use warnings;
 use testapi;
-use utils;
+use utils qw(systemctl zypper_call);
+use Utils::Systemd 'disable_and_stop_service';
```
Comment (foursixnine) [inline tests/console/cups.pm]: This module is exporting by default and it only has one thing so far, I guess it's safe to simply `use Utils::Systemd;`. However I leave this to you :) It's not really harmful to just import what we need :)

File: lib/Utils/Systemd.pm [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+our @EXPORT = qw(
+  disable_and_stop_service
+);
+
+
+=head2 disable_and_stop_service
+
+    disable_and_stop_service($service_name[, mask_service => $mask_service][, ignore_failure => $ignore_failure]);
+
+Disable and stop the service I<$service_name>.
```
Comment (foursixnine) [inline lib/Utils/Systemd.pm]: Disables and stops the specified service via C<$service_name>. (preferably use C<foo> for code)

File: lib/Utils/Systemd.pm [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+Mask it if I<$mask_service> evaluates to true. Default: false
+Raise a failure if I<$ignore_failure> evaluates to true. Default: false
+=cut
+sub disable_and_stop_service {
+    my ($service_name, %args) = @_;
```
Comment (foursixnine) [inline lib/Utils/Systemd.pm]: die if service is empty or undefined (or does the systemctl subroutine take care of it?)

File: lib/consoletest.pm (hunk 6 above)
Comment (SergioAtSUSE) [reply lib/consoletest.pm]: Done

File: tests/console/cups.pm (hunk 7 above)
Comment (SergioAtSUSE) [reply tests/console/cups.pm]: I prefer to have the function list explicitly. It is easier to remove the use statement when we remove a function call.

File: lib/Utils/Systemd.pm (hunk 8 above)
Comment (SergioAtSUSE) [reply lib/Utils/Systemd.pm]: Replaced.

File: lib/Utils/Systemd.pm (hunk 9 above)
Comment (SergioAtSUSE) [reply lib/Utils/Systemd.pm]: Done.

**Comment by SergioAtSUSE** [comment]
Self-merging because @foursixnine approved.

**Pull request merged**